### ISSUE TITLE
UIWebView crash on Swift 2 with AFNetworking 2.6.0

### ISSUE THREAD

**Issue by jakubvano** [AFNetworking/AFNetworking#3012]
When upgrading to `Swift2` and `AFNetworking` `2.6.0` method `loadRequest(progress:success:failure)` causes `EXC_BAD_ACCESS`crash.

Example:
```
override func viewDidLoad() {
        super.viewDidLoad()

        let url = NSURL(string: "https://www.google.com")
        let request =  NSURLRequest(URL: url!)
        
        webView.loadRequest(
            request,
            progress: nil,
            success: { response, html in
                return html
            },
            failure: nil)
    }
```

It seems that issue is caused by bug in `Swift2` and `Obj-C` interoperability described here: http://stackoverflow.com/questions/32837490/crash-when-calling-obj-c-method-with-nullability-annotations-from-swift

**Comment by kcharwood** [comment]
That's interesting.

So the stackoverflow post suggest making the return string nullable.

```
- (void)loadRequest:(NSURLRequest *)request
           progress:(nullable void (^)(NSUInteger bytesWritten, long long totalBytesWritten, long long totalBytesExpectedToWrite))progress
            success:(nullable NSString  * _Nullable (^)(NSHTTPURLResponse *response, NSString *HTML))success
            failure:(nullable void (^)(NSError *error))failure;
````

Does that fix the crash? Has a radar been filed?

**Comment by infimache** [comment]
Yep, making the return string nullable fixes the crash.

rdar://22911182

**Comment by kcharwood** [comment]
We don't have plans to update the 2.x line anymore, so I'm going to close this out. Feel free to put up a PR on the 3.0 branch if this is still causing an issue, and we can review it there.

:beers:

**Comment by jakubvano** [comment]
FYI, issue has been resolved with Xcode 7.3.1
